### PR TITLE
Fix build without sound feature

### PR DESCRIPTION
**Pull request by CzBiX**
#### Summary
Build "Fix build without sound feature"

#### Purpose of change

Fix the build require `sdl2-mixer` when `make SOUND=0`.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -281,14 +281,14 @@ jobs:
       - name: Build CDDA (linux)
         if: runner.os == 'Linux' && matrix.mxe == 'none' && matrix.android == 'none'
         run: |
-          make -j$((`nproc`+0)) TILES=${{ matrix.tiles }} SOUND=${{ matrix.tiles }} RELEASE=1 LOCALIZE=1 LANGUAGES=all BACKTRACE=0 PCH=0 bindist
+          make -j$((`nproc`+0)) TILES=${{ matrix.tiles }} SOUND=${{ matrix.sound }} RELEASE=1 LOCALIZE=1 LANGUAGES=all BACKTRACE=0 PCH=0 bindist
```
Comment (Qrox) [inline .github/workflows/release.yml:284]: I think the `sound` flag is supposed to enable shipping soundpacks with the game, not enabling the sound feature, see #57138.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -281,14 +281,14 @@ jobs:
...
           mv cataclysmdda-0.F.tar.gz cdda-${{ matrix.artifact }}-${{ needs.release.outputs.timestamp }}.tar.gz
       - name: Build CDDA (windows mxe)
         if: matrix.mxe != 'none'
         env:
           PLATFORM: /opt/mxe/usr/bin/${{ matrix.mxe }}-w64-mingw32.static.gcc11-
         run: |
-          make -j$((`nproc`+0)) CROSS="${PLATFORM}" TILES=1 SOUND=1 RELEASE=1 LOCALIZE=1 LANGUAGES=all BACKTRACE=1 PCH=0 bindist
+          make -j$((`nproc`+0)) CROSS="${PLATFORM}" TILES=${{ matrix.tiles }} SOUND=${{ matrix.sound }} RELEASE=1 LOCALIZE=1 LANGUAGES=all BACKTRACE=1 PCH=0 bindist
```
Comment (Qrox) [inline .github/workflows/release.yml:291]: Windows builds always have tiles enabled. If you make this change please also add the tiles flag to the build parameters above (line 88-147).

File: .github/workflows/release.yml (hunk 1 above)
Comment (CzBiX) [reply .github/workflows/release.yml:292]: I don't get it. Why we need to include the sound feature when there is no sound packs.

File: .github/workflows/release.yml (hunk 2 above)
Comment (CzBiX) [reply .github/workflows/release.yml:299]: Fixed.

File: .github/workflows/release.yml (hunk 1 above)
Comment (Qrox) [reply .github/workflows/release.yml:292]: Because the player can download a different soundpack themselves afterwards

Comment (CzBiX) [reply .github/workflows/release.yml:292]: ok, seems this PR is not useful.😅

Comment (Qrox) [reply .github/workflows/release.yml:292]: The changes to Makefile are still useful though in case someone wants to compile with sound off themselves.

Comment (CzBiX) [reply .github/workflows/release.yml:292]: I see. That will require two separate options for sound feature and sound pack, and should be in another PR.

**Pull request closed without merging**